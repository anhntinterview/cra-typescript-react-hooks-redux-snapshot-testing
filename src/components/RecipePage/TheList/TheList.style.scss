.list-wrapper {
  position: relative;
  .item {
    position: relative;
    margin-bottom: 2rem;
    h3 {
      width: 80%;
    }
  }
  button {
    position: absolute;
    top: 20px;
    right: 10px;
  }
}

.shadow-lg {
  --tw-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
    0 4px 6px -2px rgba(0, 0, 0, 0.05);
  box-shadow: var(--tw-shadow);
}

.button-style {
  color: #fff;
  font-size: 12px;
  font-weight: bold;
  text-shadow: 0 1px 0 #c04131;
  border: 1px solid #000;
  outline: none;
  border-radius: 2px;
  border-color: #dd4b39;
  padding: 5px 10px;
  background: linear-gradient(#dd4b39, #d14836);
  cursor: pointer;
  &:hover {
    border-color: #af301f;
    background: linear-gradient(#dd4b39, #c53727);
  }
  &:focus {
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.2);
    background: linear-gradient(#d74736, #ad2719);
  }
}

.uppercase {
  text-transform: uppercase;
}

.p-4 {
  padding: 1rem;
}

.px-4 {
  padding-left: 1rem;
  padding-right: 1rem;
}

.py-2 {
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

.mt-2 {
  margin-top: 0.5rem;
}

.my-1 {
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.font-semibold {
  font-weight: 600;
}

.rounded-lg {
  border-radius: 0.5rem;
}

.tracking-wide {
  letter-spacing: 0.025em;
}

.text-gray-700 {
  --tw-text-opacity: 1;
  color: rgba(55, 65, 81, var(--tw-text-opacity));
}

.text-gray-500 {
  --tw-text-opacity: 1;
  color: rgba(107, 114, 128, var(--tw-text-opacity));
}
