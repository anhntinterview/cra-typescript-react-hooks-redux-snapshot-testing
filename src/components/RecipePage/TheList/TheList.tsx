import React, { useEffect, useContext, useState } from "react";
import {
  getRecipes,
  deleteRecipe,
} from "../../../middleware/recipe.middleware";
import { RecipeContext } from "../../../contexts/RecipeContext";
import { ELEMENT_TXT } from "../../../constants/recipe.const";
import { Recipes } from "../../../types/recipe.type";
import Pagination from "./Pagination";

import "./TheList.style.scss";

export interface TheListProps {}

export function handleCurrenPage(
  setCurrentPage: React.Dispatch<React.SetStateAction<number>>
) {
  return (param: number) => {
    setCurrentPage(param);
  };
}

export function handleDelete(
  dispatch: React.Dispatch<any>,
  id: number,
  currentPage: number,
  totalPages?: number,
  recipes?: Recipes[]
) {
  return (event: React.MouseEvent<HTMLElement>) => {
    event.preventDefault();
    deleteRecipe(dispatch, id, currentPage, totalPages, recipes);
  };
}

const TheList: React.FC<TheListProps> = () => {
  const [currentPage, setCurrentPage] = useState<number>(1);
  const { recipeState, dispatch } = useContext(RecipeContext);

  const { result, status } = recipeState;

  useEffect(() => {
    getRecipes(dispatch, currentPage, undefined);
  }, [dispatch, currentPage]);

  if (status === "done") {
    const { recipes, totalPages } = result;

    return (
      <div className="list-wrapper">
        {recipes.map((item, index) => (
          <div className="item shadow-lg p-4 rounded-lg" key={index}>
            <h3 className="tracking-wide text-gray-700 mt-2">{item.title}</h3>
            <p className="text-gray-500 my-1">{item.description}</p>
            <button
              className="button-style"
              onClick={handleDelete(
                dispatch,
                item.id,
                currentPage,
                undefined,
                recipes
              )}
            >
              {ELEMENT_TXT.DELETE_BUTTON}
            </button>
          </div>
        ))}
        <Pagination
          currentPage={currentPage}
          totalPages={totalPages}
          handleCurrenPage={handleCurrenPage(setCurrentPage)}
        />
      </div>
    );
  }

  return null;
};

export default TheList;
