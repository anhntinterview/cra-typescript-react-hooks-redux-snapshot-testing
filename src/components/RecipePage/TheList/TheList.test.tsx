import * as React from "react";
import { 
  shallow, 
} from "enzyme";

import TheList from "./TheList";

describe('TheList', () => {
  it('should match a snapshot', () => {
    expect(shallow(<TheList />)).toMatchSnapshot();
  })
})
