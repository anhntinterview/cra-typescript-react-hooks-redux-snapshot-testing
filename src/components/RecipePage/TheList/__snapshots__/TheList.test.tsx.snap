// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheList should match a snapshot 1`] = `""`;
